File: capybara-rspec/pages/passenger_page.rb
require File.join(File.dirname(__FILE__), "abstract_page.rb")

class PassengerPage < AbstractPage

  def initialize()
    super("") #
  end

  def click_next()
    click_button("Next")
  end

  def enter_first_name(passenger_first_name)
    fill_in 'passengerFirstName', :with => passenger_first_name
  end

  def enter_last_name(passenger_last_name)
    fill_in 'passengerLastName', :with => passenger_last_name
  end

end

File: performance-comparision/pages/passenger_page.js
const AbstractPage = require('./abstract_page');

var webdriver = require('selenium-webdriver'),
    By = webdriver.By,
    until = webdriver.until;

class PassengerPage extends AbstractPage {

  constructor(driver) {
    super(driver);
  }

  async enterFirstName(first_name) {
    await this.driver.fill("input[name='passengerFirstName']", first_name);    
  }

  async enterLastName(last_name) {
		await this.driver.fill("input[name='passengerLastName']", last_name);
  }

  async clickNext() {
    await this.driver.click("input:has-text('Next')");
  }
};

module.exports = PassengerPage;
